src/common/codeine/db/mysql/DbUtils.java

```diff
@@ -1,187 +1,254 @@
-	{
-		ResultSet rs = null;
-		Connection connection = getConnection(false);
-		try {
-			PreparedStatement preparedStatement = connection.prepareStatement(sql, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_UPDATABLE);
-			rs = preparedStatement.executeQuery();
-			while (rs.next()) {
-				function.apply(rs);
-			}
-		} catch (SQLException e) {
-			throw prepareException(sql, connection, e, null);
-		} finally {
-			closeResultSet(rs);
-			dbConnection.closeConnection(connection);
-		}
-	}
-
-	public int executeUpdate(String sql, String... args) {
-		return executeUpdate(sql, false, args);
-	}
-	private int executeUpdate(String sql, boolean root, String... args) {
-		PreparedStatement preparedStatement = null;
-		Connection connection = root ? getConnectionForRoot(true) : getConnection(true);
-		try {
-			preparedStatement = connection.prepareStatement(sql);
-			for (int i = 1; i <= args.length; i++) {
-				preparedStatement.setString(i, args[i-1]);
-			}
-			return preparedStatement.executeUpdate();
-		} catch (SQLException e) {
-			throw prepareException(sql, connection, e, args);
-		} finally {
-			closeStatement(preparedStatement);
-			dbConnection.closeConnection(connection);
-		}
-	}
-	
-	public int executeUpdateAsRoot(String sql) {
-		return executeUpdate(sql, true);
-	}
-
-	private Connection getConnection(boolean useCompression) {
-		String url = "jdbc:mysql://"+hostSelector.mysql().host()+":" + hostSelector.mysql().port() + "/" + MysqlConstants.DB_NAME + "?connectTimeout=60000&socketTimeout=90000";
-		if (useCompression) {
-			url += "&useCompression=true";
-		}
-		try {
-			return DriverManager.getConnection(url, hostSelector.mysql().user(), hostSelector.mysql().password());
-		} catch (SQLException e) {
-			throw new ConnectToDatabaseException(url, e);
-		}
-	}
-	private Connection getConnectionForRoot(boolean useCompression) {
-		String url = "jdbc:mysql://localhost:" + hostSelector.mysql().port() + "/" + MysqlConstants.DB_NAME + "?user=root&" + "createDatabaseIfNotExist=true";
-		if (useCompression) {
-			url += "&useCompression=true";
-		}
-		try {
-			return DriverManager.getConnection(url);
-		} catch (SQLException e) {
-			throw new ConnectToDatabaseException(url, e);
-		}
-	}
-
-	@Override
-	public String toString() {
-		return "DbUtils [" + server() + "]";
-	}
-
-	public String server() {
-		MysqlConfigurationJson mysql = hostSelector.mysql();
-		String hostPort = mysql == null ? "null" : mysql.hostPort();
-		return  hostPort;
-	}
+public class DbUtils {
+
+    private final GenericObjectPoolConfig genericObjectPoolConfig = new GenericObjectPoolConfig();
+    private static Logger log = Logger.getLogger(DbUtils.class);
+    private DBConnection dbConnection = new DBConnection();
+    private Map<String, String> connectionsPool = Maps.newConcurrentMap();
+
+    @Inject
+    private MysqlHostSelector hostSelector;
+
+
+    public DbUtils() {
+        super();
+        genericObjectPoolConfig.setMaxTotal(2);
```

Comment:
is there a way to get the numbers from configuration?

Reply:
no

Reply:
fixed